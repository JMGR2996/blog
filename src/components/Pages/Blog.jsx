import React from 'react'
import { connect } from 'react-redux'
import Posts from '../Templates/Posts/Posts'
import NewPost from '../Organisms/NewPost/NewPost'

const Blog = ({ posts }) => {
  return (
    <div className='l-container'>
      <NewPost />
      <Posts posts={posts} />
    </div>
  )
}

const mapStateToProps = ({ auth }) => ({ auth })

export default connect(mapStateToProps)(Blog)
