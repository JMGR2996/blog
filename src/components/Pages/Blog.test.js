import React from 'react'
import { shallow, configure } from 'enzyme'
import Blog from './Blog'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store'
configure({ adapter: new Adapter() })

const mockStore = configureStore()

describe('Render Blog', () => {
  it('render correctly Blog component', () => {
    const store = mockStore({ posts: {} })

    const BlogComponent = shallow(<Blog store={store} />).dive()
    expect(BlogComponent).toMatchSnapshot()
  })
})
