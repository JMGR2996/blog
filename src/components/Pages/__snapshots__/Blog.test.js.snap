// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Blog render correctly Blog component 1`] = `
<div
  className="l-container container container--auth"
>
  <Connect(NewPost) />
  <Connect(Posts) />
</div>
`;
